terraform-provider-constellation/internal/provider/image_data_source.go

```diff
@@ -252,9 +252,10 @@ func (d *ImageDataSource) Read(ctx context.Context, req datasource.ReadRequest,
 
 	// Save data into Terraform state
 	diags := resp.State.SetAttribute(ctx, path.Root("image"), imageAttribute{
-		Reference: imageRef,
-		Version:   imageSemver,
-		ShortPath: apiCompatibleVer.ShortPath(),
+		Reference:        imageRef,
+		Version:          imageSemver,
+		ShortPath:        apiCompatibleVer.ShortPath(),
+		MarketplaceImage: data.MarketplaceImage.ValueBoolPointer(),
```

Comment:
good catch! I wonder if we should start only using constructor for structs to ensure that all necessary attributes are always defined.